.github/workflows/go.yml

```diff
@@ -10,6 +10,11 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    # This ensures that previous jobs for the PR are canceled when the PR is
+    # updated.
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.head_ref }}
+      cancel-in-progress: true
```

Comment:
generally should be at the top

```
# This ensures that previous jobs for the PR are canceled when the PR is
# updated.
concurrency:
  group: ${{ github.workflow }}-${{ github.head_ref }}
  cancel-in-progress: true
```

refer here https://twitter.com/kubesploit/status/1477613591465394177